internal/store/endpointslice.go

```diff
@@ -74,6 +74,56 @@ func endpointSliceMetricFamilies(allowAnnotationsList, allowLabelsList []string)
 				}
 			}),
 		),
+		*generator.NewFamilyGeneratorWithStability(
+			"kube_endpointslice_endpoints_hints",
+			"Topology routing hints attached to endpoints",
+			metric.Gauge,
+			basemetrics.ALPHA,
+			"",
+			wrapEndpointSliceFunc(func(e *discoveryv1.EndpointSlice) *metric.Family {
+				m := []*metric.Metric{}
+				for _, ep := range e.Endpoints {
+					// Hint is populated when the endpoint is configured to be zone aware and preferentially route requests to its local zone.
+					// If there is no hint, skip this metric
+					if ep.Hints != nil && len(ep.Hints.ForZones) > 0 {
+						var (
+							labelKeys,
+							labelValues []string
+						)
+
+						if ep.Hostname != nil {
+							labelKeys = append(labelKeys, "hostname")
+							labelValues = append(labelValues, *ep.Hostname)
+						}
+
+						// Did you know that k8s will "assign" endpoints from a zone
+						// This field is useful for debugging weird network routing issues
+						if ep.Zone != nil {
+							labelKeys = append(labelKeys, "endpoint_zone")
+							labelValues = append(labelValues, *ep.Zone)
+						}
+
+						// Per Docs.
+						// This must contain at least one address but no more than
+						// 100. These are all assumed to be fungible and clients may choose to only
+						// use the first element. Refer to: https://issue.k8s.io/106267
+						labelKeys = append(labelKeys, "address")
+						labelValues = append(labelValues, ep.Addresses[0])
```

Comment:
addresses are already part of `kube_endpointslice_endpoints`, I don't think we should duplicate this info

Reply:
We need a unique identifier for the endpoint. With address you can join them together and get the information from that label. Without it, you can't connect the hint with the endpoint.

Reply:
I previously thought that the hostname would be the unique identifier. If that's the address, then we should remove the hostname in favor of the address

Reply:
ok, I'll remove the hostname.